### DISCUSSION TITLE
Feature Request: thread_id

### DISCUSSION THREAD

**Discussion by breezewish** [slog-rs/slog#313] [Q&A]
It would be great if the thread_id could be printed out, which could be helpful to learn the context in multi-threaded applications.

**Comment by Techcable** [comment]
I believe this should be possible through the use of the [`FnValue` struct](https://docs.rs/slog/latest/slog/struct.FnValue.html).

The following code should create a new child logger. The new child will have a "thread_id" key in its context, computed by the `FnValue"` Because the `FnValue` is lazily evaluated )only at the point logging actually occurs), you can even clone the child logger and use it from multiple threads.

```rust
let child = parent.new(o!(
    "thread_id" => slog::FnValue(|| {
        format!("{:?}", std::thread::current().id())
    })
));
// any use of the `child` logger will have `thread_id` as context.
play(&child);
```

You should also be able to modify this to work with the root loger (just move it to the initial `o!()` instead of creating a new child).

Please let me know if this doesn't work
